Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/CosmosServerlessSparkBatchJob.java

```diff
@@ -0,0 +1,235 @@
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class CosmosServerlessSparkBatchJob extends SparkBatchJob {
+    private static final String REST_SEGMENT_SPARK_BATCH_JOB = "/activityTypes/spark/batchJobs";
+    @NotNull
+    private final AzureSparkServerlessAccount account;
+    @NotNull
+    private String jobId;
+    @NotNull
+    private final Deployable jobDeploy;
+    @NotNull
+    private final AzureHttpObservable http;
+    @NotNull
+    private final String apiVersion = ApiVersion.VERSION;
+
+    public CosmosServerlessSparkBatchJob(@NotNull AzureSparkServerlessAccount account,
+                                         @NotNull Deployable jobDeploy,
+                                         @NotNull CreateSparkBatchJobParameters submissionParameter,
+                                         @NotNull SparkBatchSubmission sparkBatchSubmission,
+                                         @NotNull Observer<AbstractMap.SimpleImmutableEntry<MessageInfoType, String>> ctrlSubject) {
+        super(submissionParameter, sparkBatchSubmission, ctrlSubject);
+        this.account = account;
+        this.jobId = UUID.randomUUID().toString();
+        this.jobDeploy = jobDeploy;
+        this.http = new AzureDataLakeHttpObservable(account.getSubscription().getTenantId(), apiVersion);
+    }
+
+    @NotNull
+    public AzureHttpObservable getHttp() {
+        return http;
+    }
+
+    @NotNull
+    public CreateSparkBatchJobParameters getSubmissionParameter() {
+        return (CreateSparkBatchJobParameters) super.getSubmissionParameter();
+    }
+
+    // WARNING: This method should only be called for Unit Test
+    public void setJobId(@NotNull String jobId) {
+        this.jobId = jobId;
+    }
+
+    @NotNull
+    public String getJobId() {
+        return jobId;
+    }
+
+    /**
+     * Create Serverless Spark batch job
+     */
+    @NotNull
+    public CreateSparkBatchJob prepareCreateSparkBatchJob(CreateSparkBatchJobParameters parameters) {
+        return new CreateSparkBatchJob()
+                // TODO: Replace name with a meaningful name
+                .withName("ServerlessSession_" + jobId)
+                .withProperties(parameters);
+    }
+
+    @NotNull
+    public Observable<com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkBatchJob> createSparkBatchJobRequest(@NotNull String jobId, @NotNull CreateSparkBatchJobParameters parameters) {
+        String url = getConnectUri().toString() + "/" + jobId;
+
+        CreateSparkBatchJob putBody = prepareCreateSparkBatchJob(parameters);
+
+        String json = putBody.convertToJson()
+                .orElseThrow(() -> new IllegalArgumentException("Bad Spark Batch Job arguments to put"));
+
+        StringEntity entity = new StringEntity(json, StandardCharsets.UTF_8);
+        entity.setContentType("application/json");
+
```

Comment:
Could you log the `PUT` and `DELETE` requests?

Reply:
Won't fix in this PR

---
In reply to: [234894059](https://github.com/Microsoft/azure-tools-for-java/pull/2367#discussion_r234894059) [](ancestors = 234894059)